### DISCUSSION TITLE
RNAseq data within seqr

### DISCUSSION THREAD

**Discussion by bellmartin** [broadinstitute/seqr#3548] [General]
Hi seqr team,
I would be grateful if you could let me know when the new features for the integration of  RNA-seq data within seqr will be made available for groups with their own seqr instance ? 

We have our own pipelines using Fraser ( and now testing Fraser2) as well as outrider and are keen to have these RNA-seq results imported into our own seqr instance.  It would be very helpful if you could let us know what format the data (Fraser & outrider results ) should have for ingesting into seqr, so we can begin formatting our data appropriately. Are you planning on releasing your own pipelines for RNA-seq analysis ?

We are happy to be used as a testing site if there are any features you would like feedback on.  We have a reasonable cohort of RNAseq samples ( ~85 mRNA samples,  majority one tissue type, 4 tissue types in total) with corresponding WGS samples.
 
Thanks in advance for these new features! 

Kind regards,
Katrina

**Comment by hanars** [comment]
@bw2 has developed the pipeline we use for RNA calling, and can speak to when we anticipate that being available

**Comment by bw2** [comment]
@bellmartin , seqr supports 4 RNA-seq related data types:

gene expression (TPM) 
gene expression outliers (such as from OUTRIDER)
splice junction outliers (such as from FRASER)
IGV.js splice junction tracks (https://github.com/igvteam/igv.js/pull/1019)

@hanars are all of these ready for external use, and what's the best way to document the input formats?

**Reply by hanars** [reply]
splice junction outliers are not ready quite yet, we should hold off until we have FRASER2 available in our own seqr instance and have finished any tweaks to the data model or the UI, but everything else is ready

**Reply by bellmartin** [reply]
Thanks @hanars  
Our current seqr version is seqr v1.0-88900306, with our code base last merged with your master as of 2023-06-24 (commit c200fdc3eee153efde298027946099be92cde59c).  We have Outlier, TPM and SpliceOutlier available to select as data types in this version of seqr.  Is this still to be updated with new features for these 3 data types or is this fully enabled as is ?   Thanks for your help.

**Reply by hanars** [reply]
SpliceOutlier has not yet been fully tested by our team, so I might hold off on that if I were you until we can iron out any issues that may still arise with it, although if you do want to try it out it theoretically should work. Outlier and TPM should work fine on the version you are running

**Comment by bw2** [comment]
As far as pipelines, we've implemented FRASER and OUTRIDER on the Hail Batch cloud compute environment. I'm not sure how useful this is for external groups. Also, the process of generating tables for upload into seqr requires annotating the RNA-seq samples with their seqr project and sample id. Unfortunately that step relies on systems that are not externally accessible.

**Reply by hanars** [reply]
I think as a lightweight MVP for this, we can document what the file format we expect is, and how to call the data, and include a note that groups are responsible for manually adding the project ID and sample ID.

**Comment by bw2** [comment]
Here's a summary of columns in the RNA-seq TSV tables that data managers can upload via: https://seqr.broadinstitute.org/data_management/rna_seq

**Normalized gene expression table (TPMs from STAR or RNAseqQC):**

<img width="324" alt="image" src="https://github.com/broadinstitute/seqr/assets/6240170/884931b4-90ab-4de9-a17c-ac8ceb58f619">


**Gene expression outliers (from a tool like OUTRIDER)**

<img width="324" alt="image" src="https://github.com/broadinstitute/seqr/assets/6240170/13f425a1-652c-4914-a98a-ddb19841b548">


**Splicing outliers (from FRASER or similar tools)**

<img width="459" alt="image" src="https://github.com/broadinstitute/seqr/assets/6240170/b433c390-d3f6-4c6e-be0c-8de4b888f757">

**Reply by hanars** [reply]
@bw2 can you also share (at least minimal) documentation for how the data was generated?

**Reply by bellmartin** [reply]
Thanks @bw2 that is really helpful!  I have a few follow up questions if you don't mind.

- For the tissue type, are there specific tissue type names that are accepted or is this an open field ?  

-  For outrider and fraser data, I assume you load the whole data set for plotting purposes? Not just the top hits ie adj.p value <1 . Correct? 

- For the ensembl geneID's,  have you encountered  any issues with different versions of gene annotation within seqr ?   
Thanks

**Comment by bw2** [comment]
The output of FRASER and OUTRIDER directly map to these columns. 
In the splicing outlier table the `rareDiseaseSamplesWithJunction` and `rareDiseaseSamplesTotal` are places to optionally add extra info about how frequently a junction is seen in other samples in the cohort and/or GTEx. These should be added to the FRASER output table or left blank.

**Comment by bw2** [comment]
I currently get the gene expression table (TPMs) by combining RNAseqQC2 per-sample output tables.

**Comment by AlessandroLas** [comment]
Hi all,
Also for me it is unclear if there are specific tissue type names that are accepted.
I tried with "blood", "whole blood" and the UBERON code for the whole blood but it won't work.

Many thanks in advance.
Alessandro.
